include/btree/chunk_ops.h

```diff
@@ -0,0 +1,129 @@
+/*-------------------------------------------------------------------------
+ *
+ * chunk_ops.h
+ *		OrioleDB abstract page chunk access API.
+ *
+ * Copyright (c) 2024, Oriole DB Inc.
+ *
+ * IDENTIFICATION
+ *	  contrib/orioledb/include/btree/chunk_ops.h
+ *
+ *-------------------------------------------------------------------------
+ */
+#ifndef __BTREE_CHUNK_OPS_H__
+#define __BTREE_CHUNK_OPS_H__
+
+#include "btree/btree.h"
+#include "btree/page_contents.h"
+
+typedef enum BTreeChunkOperationType
+{
+	BTreeChunkOperationInsert,
+	BTreeChunkOperationUpdate,
+	BTreeChunkOperationDelete
+} BTreeChunkOperationType;
+
+typedef enum BTreeChunkType
+{
+	BTreeChunkHiKeys,
+	BTreeChunkNonLeaf,
+	BTreeChunkLeaf
+} BTreeChunkType;
+
+typedef struct BTreeChunkBuilderData
+{
+	BTreeDescr *treeDesc;
+
+	Pointer		builderData;
+	Size		builderMaxSize;
+	Size		builderSize;
+
+	BTreeChunkType chunkType;
+} BTreeChunkBuilderData;
+
+typedef struct PartialPageState PartialPageState;
+
+typedef struct BTreeChunkDesc BTreeChunkDesc;
+
+typedef struct BTreeChunkOps
+{
+	/*
+	 * Estimate the new chunk size after the operation.
+	 */
+	int32		(*estimate_change) (BTreeChunkDesc *desc, OffsetNumber itemOffset,
+									BTreeChunkOperationType operation,
+									Pointer tupleHeader, OTuple tuple);
+
+	/*
+	 * Perform the operation.  A caller should take care about the size of the
+	 * chunk.  Extension of the chunk should be performed before "perform_change",
+	 * and shrinking of the chunk should be performed after "perform_change".
+	 */
+	void		(*perform_change) (BTreeChunkDesc *desc, OffsetNumber itemOffset,
+								   BTreeChunkOperationType operation,
+								   Pointer tupleHeader, OTuple tuple);
+
+	/*
+	 * Compare given key to the chunk tuple.
+	 */
+	OBTreeKeyCmp keycmp;
```

Comment:
There should be a different signature. This function should compare a tuple stored in the chunk (at a given offset) to the given tuple.

Reply:
Added a separate implementation of tuple comparison within the chunk.